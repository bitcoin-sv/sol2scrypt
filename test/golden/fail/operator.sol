pragma solidity ^0.8.10;

contract SolidityTest {
  
    function arithmetic(uint16 a, uint16 b) public {
        int128 ff = -2 ** 127 ;
    }

}
