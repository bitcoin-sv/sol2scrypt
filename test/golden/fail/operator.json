[
    {
        "src": "test/golden/fail/operator.sol:6:25:6:26",
        "type": "Error",
        "message": "unsupported `&` on non-bytes expression"
    },
    {
        "src": "test/golden/fail/operator.sol:8:24:8:25",
        "type": "Error",
        "message": "unsupported `|` on non-bytes expression"
    },
    {
        "src": "test/golden/fail/operator.sol:10:25:10:26",
        "type": "Error",
        "message": "unsupported `^` on non-bytes expression"
    },
    {
        "src": "test/golden/fail/operator.sol:17:11:17:13",
        "type": "Error",
        "message": "unsupported `|=` on non-bytes expression"
    },
    {
        "src": "test/golden/fail/operator.sol:18:11:18:13",
        "type": "Error",
        "message": "unsupported `^=` on non-bytes expression"
    },
    {
        "src": "test/golden/fail/operator.sol:19:11:19:13",
        "type": "Error",
        "message": "unsupported `&=` on non-bytes expression"
    },
    {
        "src": "test/golden/fail/operator.sol:26:22:26:23",
        "type": "Error",
        "message": "unsupported `~` on non-bytes expression"
    },
    {
        "src": "test/golden/fail/operator.sol:30:24:30:26",
        "type": "Error",
        "message": "unsupported binary operator `**`"
    }
]